primitives/src/proposition/mod.rs

```diff
@@ -126,23 +122,17 @@ where
     NotProposition::new(proposition)
 }
 
-/// It verifies if the Identifier in the context has got a valid `InvestorZKProof` and its
-/// associate `CustomDueDiligence`.
-#[inline]
-pub fn has_valid_proof_of_investor(ticker: Ticker) -> ValidProofOfInvestorProposition {
-    ValidProofOfInvestorProposition { ticker }
-}
-
 /// Helper function to run propositions from a context.
 pub fn run(condition: &Condition, context: &Context) -> bool {
     match condition.condition_type {
         ConditionType::IsPresent(ref claim) => exists(claim).evaluate(context),
         ConditionType::IsAbsent(ref claim) => not(exists(claim)).evaluate(context),
         ConditionType::IsAnyOf(ref claims) => any(claims).evaluate(context),
         ConditionType::IsNoneOf(ref claims) => not(any(claims)).evaluate(context),
-        ConditionType::HasValidProofOfInvestor(ref ticker) => {
-            has_valid_proof_of_investor(ticker.clone()).evaluate(context)
-        }
+        ConditionType::HasValidProofOfInvestor(..) => context
+            .claims
+            .iter()
+            .any(|claim| exists(claim).evaluate(context)),
```

Comment:
Can we do something like `any(claims).evaluate(context)` to keep the same format as the rest?

Reply:
We don't have the claim or claim list in this case so it is not possible to do so. even the current code is also wrong we are iterating the same claim vector and matching with the same. so it doesn't make sense. 

It only needs to check whether the length of the claims vector should not be zero to verify the existence of the scope claim.